CMakeLists.txt

```diff
@@ -1282,28 +1282,43 @@ elseif(UNIX AND NOT APPLE AND NOT ANDROID AND NOT RISCOS AND NOT HAIKU)
       set(HAVE_INOTIFY 1)
     endif()
 
-    if(PKG_CONFIG_FOUND)
-      pkg_search_module(DBUS dbus-1 dbus)
-      if(DBUS_FOUND)
-        set(HAVE_DBUS_DBUS_H TRUE)
-        include_directories(${DBUS_INCLUDE_DIRS})
-        list(APPEND EXTRA_LIBS ${DBUS_LIBRARIES})
-        # Fcitx need only dbus.
-        set(HAVE_FCITX TRUE)
-      endif()
+    # Cater for both multilib and multiarch, native and cross-compiling build
+    find_path(DBUS_H_INCLUDE_DIRS NAMES dbus/dbus.h PATH_SUFFIXES dbus-1.0)
+    if(ARCH_64)
+      set(DBUS_INC_SEARCH_PATH /usr/lib64/dbus-1.0/include /usr/lib/x86_64-linux-gnu/dbus-1.0/include)
+    else()
+      set(DBUS_INC_SEARCH_PATH /usr/lib/dbus-1.0/include /usr/lib/i386-linux-gnu/dbus-1.0/include)
+    endif()
```

Comment:
As well as making assumptions about libdbus implementation details, this also assumes everything is x86.

Reply:
This is a valid point. The patch was originally made for my project with subset of target platforms in mind.